### PR TITLE
[CircleCI] Add Python3.8-gcc9 config

### PR DESCRIPTION
**Pull request by malfet**
`pytorch-linux-bionic-py3.8-gcc9` is based on Ubuntu 18.04 using gcc-9 and python-3.8
`pytorch-linux-bionic-cuda10.2-cudnn7-py3.8-gcc9` adds CUDA-10.2 to the same configuration

Also this in this PR:
 - Updates valgrind to 3.15.0
 - Fixes bug when gcc-5.5 were used in gcc-5.4 configurations
 - Do not install `typing` when installing Python-3.8 from Conda
 - Install `llvmdev-8` to for `numba/llvmlite` package compilation to succeed

### CODE REVIEW

File: .circleci/docker/common/install_gcc.sh [hunk 1]
```diff
@@ -7,7 +7,11 @@ if [ -n "$GCC_VERSION" ]; then
   # Need the official toolchain repo to get alternate packages
   add-apt-repository ppa:ubuntu-toolchain-r/test
   apt-get update
-  apt-get install -y g++-$GCC_VERSION
+  if [ "$UBUNTU_VERSION" == "16.04" -a "$GCC_VERSION" == "5" ]; then
```
Comment (seemethere) [inline .circleci/docker/common/install_gcc.sh:10]: Prefer to use a single `=` within shell conditionals.

https://unix.stackexchange.com/a/72042/249966

**Review by seemethere — APPROVED** [review]

**Comment by malfet** [comment]
Moving adding  new build configuration to https://github.com/pytorch/pytorch/pull/38805 as there are several issues with compiling top-of-tree pytorch with gcc-9, for example https://github.com/pytorch/pytorch/pull/38802

**Review by seemethere — APPROVED** [review]

File: .circleci/docker/build.sh [hunk 2]
```diff
@@ -155,6 +157,22 @@ case "$image" in
     DB=yes
     VISION=yes
     ;;
+  pytorch-linux-bionic-py3.8-gcc9)
+    ANACONDA_PYTHON_VERSION=3.8
+    GCC_VERSION=9
+    PROTOBUF=yes
+    DB=yes
+    VISION=tes
```
Comment (ailzhang) [inline .circleci/docker/build.sh:165]: I assume this should be `yes` instead of `tes`? ;)

Comment (malfet) [reply .circleci/docker/build.sh:165]: Yes, you are right. Can you please fix the typo in PR you are working on.

Comment (ailzhang) [reply .circleci/docker/build.sh:165]: sure!

**Pull request closed without merging**